src/lib/integration_test_local_engine/swarm_network.ml

```diff
@@ -0,0 +1,557 @@
+open Core
+open Async
+open Integration_test_lib
+
+(* Parts of this implementation is still untouched from the Kubernetes implementation. This is done
+  just to make the compiler happy as work is done on the local engine.
+*)
+
+(* exclude from bisect_ppx to avoid type error on GraphQL modules *)
+[@@@coverage exclude_file]
+
+module Node = struct
+  type t =
+    { swarm_name: string
+    ; service_id: string
+    ; graphql_enabled: bool
+    ; network_keypair: Network_keypair.t option }
+
+  let id {service_id; _} = service_id
+
+  let network_keypair {network_keypair; _} = network_keypair
+
+  let base_kube_args t = ["--cluster"; t.swarm_name]
+
+  let get_container_cmd t =
+    Printf.sprintf "$(docker ps -f name=%s --quiet)" t.service_id
+
+  let run_in_postgresql_container node ~cmd =
+    let base_args = base_kube_args node in
+    let base_kube_cmd = "kubectl " ^ String.concat ~sep:" " base_args in
+    let kubectl_cmd =
+      Printf.sprintf "%s -c %s exec -i %s-0 -- %s" base_kube_cmd
+        node.service_id node.service_id cmd
+    in
+    let%bind cwd = Unix.getcwd () in
+    Util.run_cmd_exn cwd "sh" ["-c"; kubectl_cmd]
+
+  (* 
+    Maybe we get rid of this? We are redirecting all logs to stdout which the test executive will pick up
```

Comment:
the redirected logs are the ones from the container?

Reply:
Sorry, this comment is a little misleading. It was more of a reminder to myself that the logging infrastructure for the local engine will be different than the cloud engine and that this function is subject to change later down the road.